uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -63,9 +72,37 @@ def connection_lost(self, exc):
 
     def eof_received(self):
         pass
-
+    
     def data_received(self, data):
-        self.conn.receive_bytes(data)
+        try:
+            self.conn.receive_data(data)
+        except wsproto.utilities.RemoteProtocolError as err:
+            if err.event_hint is not None:
+                #self.handle_no_connect(wsproto.events.CloseConnection(code=err.event_hint.status_code, reason=str(err)))
+                output = self.conn.send(err.event_hint)
+                self.transport.write(output)
+                self.transport.close()
+            else:
+                self.handle_no_connect(wsproto.events.CloseConnection())
+            return
```

Comment:
This corresponds to "Switch from Fail events being returned to RemoteProtocolError`s being raised." in the changelog.